################################################################################
SHORT DESCRIPTION: 
################################################################################
XSA-149 (http://xenbits.xen.org/xsa/advisory-149.html)
xen: free domain's vcpu array

################################################################################
LONG DESCRIPTION: 
################################################################################
Source: http://xenbits.xen.org/xsa/advisory-149.html
Patches: xsa149.patch

A domain's primary array of vcpu pointers can be allocated by a toolstack
exactly once in the lifetime of a domain via the XEN_DOMCTL_max_vcpus
hypercall.

This array is leaked on domain teardown. This memory leak could -- over time --
exhaust the host's memory.


################################################################################
CHANGELOG 
################################################################################
Added in OpenXT, Xen 4.3.4 patch-queue: Eric Chanudet <chanudete@ainfosec.com>

################################################################################
PATCHES 
################################################################################
--- a/xen/common/domain.c
+++ b/xen/common/domain.c
@@ -751,6 +751,7 @@ static void complete_domain_destroy(stru
 
     xsm_free_security_domain(d);
     free_cpumask_var(d->domain_dirty_cpumask);
+    xfree(d->vcpu);
     free_domain_struct(d);
 
     send_global_virq(VIRQ_DOM_EXC);
